lib/puppet/util/run_mode.rb

```diff
@@ -15,6 +15,8 @@ def self.[](name)
         @run_modes ||= {}
         if Puppet::Util::Platform.windows?
           @run_modes[name] ||= WindowsRunMode.new(name)
+        elsif Puppet::Util::Platform.linux?
+          @run_modes[name] ||= LinuxRunMode.new(name)
         else
           @run_modes[name] ||= UnixRunMode.new(name)
         end
```

Comment:
Multiple run modes will require further logic here de determine which one to prefer.  For some operating systems, a simple test of the os is enough, but for Linux it will probably need something smarted, e.g. detect if AIO or not (to run in AIO RunMode), test if parent PID is 1 or not (for SystemdRunMode), etc.